File: js/upper-lower.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

// Função para ler a entrada do usuário de forma assíncrona
function askQuestion(query) {
    return new Promise(resolve => {
        readline.question(query, resolve);
    });
}

async function main() {
  // Solicitando o nome e sobrenome do usuário
  let nome = await askQuestion("Digite seu nome: ");
  let sobrenome = await askQuestion("Digite seu sobrenome: ");

  // Convertendo o primeiro caractere do nome para minúsculo e o restante para maiúsculo
  nome = nome.charAt(0).toLowerCase() + nome.slice(1).toUpperCase();

  // Convertendo o primeiro caractere do sobrenome para maiúsculo e o restante para minúsculo
  sobrenome = sobrenome.charAt(0).toUpperCase() + sobrenome.slice(1).toLowerCase();

  // Exibindo o resultado
  console.log(nome);
  console.log(sobrenome);

  // Fechando o readline
  readline.close();
}

main();

File: cpp/upper-lower.cpp
#include <iostream>
#include <string>
#include <cctype> // Para as funções tolower() e toupper()

using namespace std;

int main() {
    string nome, sobrenome;

    // Obtendo o nome e o sobrenome do usuário
    getline(cin, nome);
    getline(cin, sobrenome);

    // Convertendo o primeiro caractere do nome para minúsculo e o restante para maiúsculo
    nome[0] = tolower(nome[0]);
    for (size_t i = 1; i < nome.size(); ++i) {
        nome[i] = toupper(nome[i]);
    }

    // Convertendo o primeiro caractere do sobrenome para maiúsculo e o restante para minúsculo
    sobrenome[0] = toupper(sobrenome[0]);
    for (size_t i = 1; i < sobrenome.size(); ++i) {
        sobrenome[i] = tolower(sobrenome[i]);
    }

    // Imprimindo o nome e o sobrenome modificados
    cout << nome << endl;
    cout << sobrenome << endl;

    return 0;
}
